src/coffee/sync/product-sync.coffee

```diff
@@ -37,12 +37,20 @@ class ProductSync extends BaseSync
     super new_obj, old_obj
 
   _doMapActions: (diff, new_obj, old_obj) ->
+    # New variant can have attributes with new values and this can cause conflict if the product still
+    # has old value of the sameForAll attribute. That's why it's necessary to update the attributes
+    # before adding new variant
+    variantActions = @_mapActionOrNot 'variants', => @_utils.actionsMapVariants(diff, old_obj, new_obj)
```

Comment:
Just looking at these three lines the second example is more readable. But now to understand the second example you need to look for the function definition so reading through the code goes up and down.